### ISSUE TITLE
MPQ Editor: bug with some filenames

### ISSUE THREAD

**Issue by Magnifikator** [ladislav-zezula/StormLib#143]
As the provided support mail address for MPQ editor is not working anymore, I will try it here (sorry):

I was upgrading MPQ-Editor from 3.5.1.823 to 3.6.0.850 and found a bug
in extraction of some files with specific filenames:

Examples:

Creature\DRAENEIMALEKID\DRAENEIMALEKID_01..BLP
will be extracted new in:
Creature\DRAENEIMALEKID\DRAENEIMALEKID_01%2E.BLP

ITEM\OBJECTCOMPONENTS\WEAPON\Axe_1H_Horde_A_02Blue .blp
will be extracted new in:
ITEM\OBJECTCOMPONENTS\WEAPON\Axe_1H_Horde_A_02Blue%20.blp

By browsing the files in the editor they are shown correct.
But at least on a Windows OS these are different files.
(I'm using WotLK MPQs and Win8.1 OS.)

**Comment by ladislav-zezula** [comment]
Hi, please, send me the map which you are extracted from. It will help me to fix the issue faster.

And BTW, the e-mail address still works. Just tested.

**Comment by Magnifikator** [comment]
What do you mean with map?

I import different MPQ like that:
fullpath\MPQEditor.exe /merged enGB\base-enGB.MPQ enGB\locale-enGB.MPQ enGB\expansion-locale-enGB.MPQ enGB\lichking-locale-enGB.MPQ enGB\patch-enGB.MPQ enGB\patch-enGB-2.MPQ enGB\patch-enGB-3.MPQ common.MPQ common-2.MPQ expansion.MPQ lichking.MPQ /patch patch.MPQ patch-2.MPQ patch-3.MPQ

After that I export it.


The problem with your mail is maybe because you forward it in a strange way. That's the error message I got:
This is the mail system at host smtp1.czechia.com.
I'm sorry to have to inform you that your message could not be delivered to one or more recipients. It's attached below.
For further assistance, please send mail to postmaster.
If you do so, please include this problem report. You can
delete your own text from the attached returned message.

<zezula@volny.cz>: host cax.virusfree.cz[212.224.105.17] said: 550 Please see
    http://www.openspf.org/Why?id=admin%40landofelves.net&ip=217.198.120.38&receiver=bq
    : Reason: mechanism (in reply to MAIL FROM command)

The link at the end explains the Problem with SPF very good.

PS: I will remove this part with mail from this message in Github after you have read it ;-) Because of sensitive information.

**Comment by ladislav-zezula** [comment]
> What do you mean with map?

The MPQ you are exporting from. Are the double-dot and space-dot file names in one of the MPQs above?

**Comment by Magnifikator** [comment]
Yes, the files are parts of **common.MPQ** of wow 3.3.5 (WotLK)

**Comment by ladislav-zezula** [comment]
File name escaping was added at some point because in MPQ listfiles, the file name can have any characters (spaces, dots, invalid file name chars). I changed the escaping procedure so that it does not include space, dot and tab at the end of file name (before extension).

TLDR: Both file names you reported are now extracted as they are; without name escaping.
Download the latest MPQ editor from my website.
